### ISSUE TITLE
AHN2 + Kadastral data

### ISSUE THREAD

**Issue by romulogoncalves** [NLeSC/TEAM2018#40]
@meiertgrootes Elena proposed such work for the November sprint. Could you elaborate a bit more about needs to be achieved?

**Comment by meiertgrootes** [comment]
Hi Romulo,

The basic goal is to use the kadaster data to classify all points in AHN2 and AHN3 into, e.g. road, water, building, other. The Challenge here is to filter the point cloud data through the polygons provided in the Kadaster files as efficiently as possible to avoid excessive computational times. After a previous conversation with you and Jisk I had gathered that there is experience at the center in doing this, e.g. by use of indexing and a database. As we hadn’t started on this in the context of eEcoLiDAR yet, but it is high on the to-do list, it crossed my mind when Elena asked whether there might be a way eEcoLiDAR could benefit from the sprint.
I have the Kadaster data downloaded and/or will make sure to have it before the sprint, and the AHN data is also present, as you well know. My naïve guess was that at the very least setting up the infrastructure to process the data could be achieved during the sprint if someone with expertise in this regard were on the team. After that it would only be a question of running it on the VMs. Beyond advancing the project, it would also result in a AHN data sets with highly accurate point classifications which could be published by NLeSC as a resource, facilitating further open use of the AHN  data sets.

Cheers,
Meiert

**Comment by romulogoncalves** [comment]
@meiertgrootes @elboyran  is there project hours for this work? If yes, which one?

**Comment by elboyran** [comment]
I told you there are.
eEcoLIDAR project hours.
We need Jisk in our team.

Regards,
Elena

**Comment by romulogoncalves** [comment]
@elboyran it just to keep for the records :)
All issues got the same comment. Jisk is not joining the Sprint so any feedback from him should be collected before the Sprint.

**Comment by elboyran** [comment]
@meiertgrootes, could you, please, take Jisk's knowledge? Who else can we drag in the team? Bouwe would be good, but he's part of the DynaSlum project/team. Not sure who else has experience in this. I think we need at least a 3rd person ;-) but not sure who that could be.

**Comment by romulogoncalves** [comment]
@elboyran SatSense team will be in the same room as you so you can share knowledge. I will be available first day for brainstorming.

**Comment by elboyran** [comment]
@romulogoncalves , thanks!

**Comment by meiertgrootes** [comment]
@romulogoncalves @elboyran  Thanks! After discussion with Jisk and Stefan PostGIS seems the way to go. Access to someone with PostGIS experience would be very beneficial, but possibly given by SatSense team and @romulogoncalves on first day

**Comment by meiertgrootes** [comment]
@elboyran @fdiblen 
I've provided a list of resources below which hopefully will provide everyone with (a) A starting point for the data and tools likely to be required, (b) a starting point, building on previous NLeSC activities, for the aims of the sprint.

### Previous NLeSC Knowledge (Should be very relevant)
http://knowledge.esciencecenter.nl/content/pcdms_usage_guide.pdf

### eEcoLiDAR Github
https://github.com/eEcoLiDAR/
and in particular
https://github.com/eEcoLiDAR/miscellaneous/tree/lidar_tilling

### Databases

**PostgreSQL**

Main page 
https://www.postgresql.org

Resources/tutorial
https://www.postgresql.org/docs/
https://www.postgresql.org/docs/online-resources/


**PostGIS**

Main web page
https://postgis.net

Intro/tutorial
https://postgis.net/workshops/postgis-intro/



### Pointcloud data and tools

**Point cloud visualisation**
CloudCompare
https://www.cloudcompare.org

Fugroviewer
https://www.fugro.com/about-fugro/our-expertise/technology/fugroviewer

**Geospatial data visualisation**
QGIS 3
https://www.qgis.org/en/site/

**Pointcloud data format(s)** 

LAS/LAZ
Format definiition
https://www.asprs.org/wp-content/uploads/2010/12/LAS_1_4_r13.pdf

**Tools**

LAStools
https://rapidlasso.com

PDAL
https://pdal.io/index.html
https://pdal.io/stages/filters.overlay.html

### NL Kadaster data
https://www.pdok.nl/introductie?articleid=1976855

### AHN2 data 
https://www.pdok.nl/introductie?articleid=1948857
This data resides on the surf hosted webdav server as detailed in the eEcoLiDAR GitHub

**Comment by meiertgrootes** [comment]
@fdiblen @elboyran I've had 2 trains cancelled and will be in at ca. 09:35

**Comment by elboyran** [comment]
I have meeting with Rob until 9:30. My bus is also late

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by romulogoncalves** [comment]
@meiertgrootes were all the goals achieved? Could you summarize in 3 lines what was achieved? Can we consider it done and close this issue?

**Comment by meiertgrootes** [comment]
The goals have largely been achieved.
-PostGIS infrastructures to select and export Kadaster Polygons is in place and works. 
- Filtering using PDAL to enrich PC data has been demonstrated to work on trial data. 

Remaining issue(s): 
-point-in-polygon with PC size still slow even with limited polygons.
 Nevertheless, Calculation for all now tractable, although at significant computational expense.
 Given one-off nature of calculation investment of CPU time maybe more efficient than further      
 optimisation efforts. 
-set-up start final run for all data. Project currently requires computational resources for other calculation. Start as soon resources free.
-ditches are not provided as polygons but as lines -> consider 'broadening' ditches. Feedback from PI on requirement needed before decision
